addons/test/misra/misra-test.c

```diff
@@ -779,6 +779,13 @@ static void misra_10_8(u8 x, s32 a, s32 b) {
 int (*misra_11_1_p)(void); // 8.4
 void *misra_11_1_bad1 = (void*)misra_11_1_p; // 11.1 8.4
 
+// #12172
+typedef void (*pfFunc)(uint32_t some);
```

Comment:
it's unfortunate to write such generic type/variable names in the global scope in this misra-test.c. There are lots of variables, types and functions and we don't want to have any duplicates. Did you check in the whole file that there is no other function `f` or global array `data`. Next time you need to add an array or function what will you call it instead of `data` or `f`. 

ensure that they are more unique.

I have used the convention to append `_11_1` for names in global scope that apply to testing of rule `11.1`. So please follow that convention. pfFunc_11_1 , data_11_1 and f_11_1